src/lib/index.js

```diff
@@ -350,19 +350,25 @@ lib.noneOrAll = function(containerIn, containerOut, attrList) {
     }
 };
 
-/** merge data array into calcdata items
+lib.mergeArray = function(traceAttr, cd, cdAttr) {
```

Comment:
> :dolls: with a description of the difference between mergeArray and fillArray?

Sure. :+1: 

> Makes me wonder whether any of our other existing merge usages should be fill...

Good point. Many routines in `Drawing` have fallbacks in case a required calcdata field is missing. The fallback pattern works well when the fallback value is constant for all cases (e.g. missing items in a `marker.color` array get drawn as `#444`), but not so well when it is depends on other things (e.g. for `hoverinfo` here which depends on `gd.data.length` and `trace.type`). 

That said, replacing all `Lib.mergeArray` calls with `Lib.fillArray` should not break anything. I could do that if you prefer. It might slow things down a little bit when e.g. 

```js
trace.x.length // => 1e5
trace.marker.color.length // => 2
```
but I'd call that an edge case.

> mergeArray doesn't need fn?

Not at the moment.